src/common/data/serverActions/getUserData.tsx

```diff
@@ -0,0 +1,53 @@
+'use server'
+
+import { AppLogger } from '@/common/services/Logger/Logger'
+import { createClient } from '@/common/services/supabase/server'
+
+interface GetUserDataResponse {
+  data?: {
+    email: string
+    avatarUrl: string
+  } | null
+  error?: string
+}
+
+export const getUserData = async (): Promise<GetUserDataResponse> => {
```

Comment:
Just wanted to point out for project consistency sake that I considered this supabase database access layer logic for jobApplication as an "api" (see `features/jobTracker/data/api/jobApplications.ts`). The serverActions are moreso the wrappers around the API logic so I'd recommend making the server actions thin and extracting the API logic into a separate file that contains all the CRUD operations.

Next.js docs state:
"Server Actions are asynchronous functions that are executed on the server. They can be used in Server and Client Components to handle form submissions and data mutations in Next.js applications."

https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations

Not trying to be a stickler, just trying to stay organized. Otherwise, I look the logic here. I might try to extract it into some reusable handlers.

Reply:
sounds good to me, i'm on it